### ISSUE TITLE
🔧 IMMEDIATE: Upgrade Python Dependencies for ML Foundation

### ISSUE THREAD

**Issue by jannekbuengener** [jannekbuengener/Claire_de_Binare#196]
### Objective
Upgrade Python dependencies to support ML development pipeline without breaking existing functionality.

### Current Dependencies Analysis
Based on current requirements.txt:
- ✅ Basic Python ecosystem (flask, redis, requests)
- ❌ Missing ML/Data Science stack
- ❌ Missing time-series analysis tools
- ❌ Missing feature engineering libraries

#### Core ML Stack
`
# Data Science & ML Core
pandas>=2.1.0              # Data manipulation and analysis
numpy>=1.24.0               # Numerical computing
scikit-learn>=1.3.0         # Machine learning algorithms
scipy>=1.11.0               # Scientific computing

# Time Series & Technical Analysis
ta>=0.10.2                  # Technical analysis indicators
arch>=6.2.0                # Time series volatility models
statsmodels>=0.14.0         # Statistical models

# Feature Engineering
feature-engine>=1.6.0       # Automated feature engineering
tsfresh>=0.20.0             # Time series feature extraction

# Model Training & Optimization
optuna>=3.4.0               # Hyperparameter optimization
mlflow>=2.7.0               # ML experiment tracking
joblib>=1.3.0               # Efficient model serialization
`

#### Advanced ML (Phase 2+)
`
# Deep Learning (Optional - Phase 2)
torch>=2.1.0                # PyTorch for neural networks
transformers>=4.35.0        # Transformer models

# Specialized Libraries
xgboost>=2.0.0              # Gradient boosting
lightgbm>=4.1.0             # Light gradient boosting
catboost>=1.2.0             # Categorical boosting

# Time Series Database
influxdb-client>=1.38.0     # Time series database client
`

#### Phase 1: Core Dependencies (This Issue)
1. Add core ML dependencies to requirements.txt
2. Test compatibility with existing codebase
3. Update Docker configurations
4. Validate all services still work

#### Phase 2: Optional Dependencies (Future)
- Deep learning frameworks (conditional install)
- GPU support (if available)
- Advanced time series libraries

### Updated requirements.txt
`
# Existing dependencies (keep as-is)
aiohttp==3.12.14
cryptography==44.0.0
flask==3.1.2
psycopg2-binary==2.9.9
python-dateutil==2.8.2
python-dotenv==1.0.0
python-json-logger==2.0.7
redis==5.0.1
requests==2.32.4

# NEW: ML Foundation Dependencies
pandas>=2.1.0
numpy>=1.24.0
scikit-learn>=1.3.0
scipy>=1.11.0
ta>=0.10.2
optuna>=3.4.0
mlflow>=2.7.0
joblib>=1.3.0
feature-engine>=1.6.0

# NEW: Time Series Support
arch>=6.2.0
statsmodels>=0.14.0
influxdb-client>=1.38.0

# NEW: Development & Testing
pytest-mock>=3.12.0        # Enhanced mocking for ML tests
`

### Testing Plan
1. **Backward Compatibility**: Ensure all existing services continue to work
2. **Import Tests**: Verify all new libraries can be imported
3. **Memory Usage**: Check memory footprint impact
4. **Docker Build**: Ensure Docker images build successfully
5. **CI/CD**: Validate pipeline still passes

### Implementation Steps
1. Update requirements.txt with core ML dependencies
2. Update requirements-dev.txt if needed
3. Test locally with existing services
4. Update Docker configurations
5. Run full test suite
6. Update CI/CD if needed

### Risk Assessment
- **LOW**: Core libraries (pandas, numpy, scikit-learn) are stable
- **MEDIUM**: Some libraries may have binary dependencies
- **MITIGATION**: Staged rollout, test thoroughly before merge

### Success Criteria
- ✅ All new ML dependencies install successfully
- ✅ Existing services remain functional
- ✅ Docker builds complete without errors
- ✅ CI/CD pipeline passes
- ✅ Memory usage increase <100MB

### Follow-up Actions
After this issue:
1. Start Phase 1 ML implementation (#192)
2. Create ML services directory structure
3. Implement first feature engineering pipeline
4. Add ML monitoring to health checks

### Notes
- Keep existing version pins where critical for stability
- Use minimum version requirements for ML libraries to allow updates
- Consider creating ml-requirements.txt for optional ML-only installs

---
Priority: 🔥 IMMEDIATE
Blocks: #192, #193, #194, #195
Epic: ML Foundation Roadmap
